Client/Frontend/Theme/ThemeManager.swift

```diff
@@ -9,101 +9,72 @@ import Shared
 /// through the `themeManager` variable, while using dependency injection.
 protocol Themeable { }
 
-extension Themeable {
-    var themeSystem: ThemeManager { return ThemeManager.shared }
+extension Themeable {}
+
+enum ThemeType: String {
+    case light = "normal" // This needs to match the string used in the legacy system
+    case dark
+
+    func getInterfaceStyle() -> UIUserInterfaceStyle {
+        switch self {
+        case .light:
+            return .light
+        case .dark:
+            return .dark
+        }
+    }
+}
+
+protocol ThemeManager {
+    var currentTheme: Theme { get }
+
+    func getInterfaceStyle() -> UIUserInterfaceStyle
 }
 
 /// The `ThemeManager` will be responsible for providing the theme throughout the app
-///
-/// IMPORTANT NOTE:
-/// The current functionality of `ThemeManager` is restricted to providing the default
-/// theme. The rest of the functionality, is a sketching out of how implementing user
-/// defined themes might work. It's a work in progress and needs not be touched until
-/// such a feature is actually added.
-final class ThemeManager {
+final class DefaultThemeManager: ThemeManager {
+
+    private enum ThemeKeys {
+        static let themeName = "prefKeyThemeName"
+        static let automaticBrightness = "prefKeyAutomaticSliderValue"
+        static let systemThemeIsOn = "prefKeySystemThemeSwitchOnOff"
+        static let automaticSwitchIsOn = "prefKeyAutomaticSwitchOnOff"
+    }
 
     // MARK: Singleton
-    static let shared = ThemeManager()
 
-    // MARK: - Variables
-    private var profile: Profile!
+    static let shared = DefaultThemeManager()
 
-    /// The theme key of the current set theme. If no custom theme is set, then
-    /// this will return the `defaultThemeKey`
-    private var themeKey: String {
-        if let key = profile.prefs.stringForKey(PrefsKeys.ThemeManagerCustomizationKey) { return key }
```

Comment:
I think we could remove ThemeManagerCustomizationKey from the PrefsKeys? Seems to be not used anymore?